tardis/visualization/tools/convergence_plot.py

```diff
@@ -0,0 +1,389 @@
+    simulation.
+
+    Parameters
+    ----------
+    iteration : int
+        iteration number
+    """
+
+    def __init__(self, iterations, **kwargs):
+        self.iterable_data = {}
+        self.value_data = defaultdict(list)
+        self.iterations = iterations
+        self.current_iteration = 1
+        self.luminosities = ["Emitted", "Absorbed", "Requested"]
+        self.plasma_plot = None
+        self.luminosity_plot = None
+
+        if "colorscale" in kwargs:
+            self.colorscale = transition_colors(
+                kwargs["colorscale"], iterations=self.iterations
+            )
+        else:
+            self.colorscale = transition_colors(iterations=self.iterations)
+
+        if "plasma_plot_config" in kwargs:
+            if not isinstance(kwargs["plasma_plot_config"], dict):
+                raise TypeError("Expected dict in plasma_plot_config argument")
+            self.plasma_plot_config = kwargs["plasma_plot_config"]
+
+        if "luminosity_plot_config" in kwargs:
+            if not isinstance(kwargs["luminosity_plot_config"], dict):
+                raise TypeError(
+                    "Expected dict in luminosity_plot_config argument"
+                )
+            self.luminosity_plot_config = kwargs["luminosity_plot_config"]
+
+    def fetch_data(self, name=None, value=None, item_type=None):
+        """
+        This allows user to fetch data from the Simulation class.
+        This data is stored and used when an iteration is completed.
+
+        Parameters
+        ----------
+        name : string
+            name of the data
+        value : string or array
+            string or array of quantities
+        item_type : string
+            either iterable or value
+
+        """
+        if item_type == "iterable":
+            self.iterable_data[name] = value
+        if item_type == "value":
+            self.value_data[name].append(value)
+
+    def create_plasma_plot(self):
+        """
+        Creates an empty plasma plot.
+        The default layout can be overridden by passing plasma_plot_config dictionary in the run_tardis function.
+        """
+        fig = go.FigureWidget().set_subplots(rows=1, cols=2, shared_xaxes=True)
+        fig.add_scatter(row=1, col=1)
+        fig.add_scatter(row=1, col=2)
+
+        fig = fig.update_layout(
+            xaxis={
+                "tickformat": "g",
+                "title": r"$Shell~~Velocity$",
+            },
+            xaxis2={
+                "tickformat": "g",
+                "title": r"$Shell~~Velocity$",
+                "matches": "x",
+            },
+            yaxis={
+                "tickformat": "g",
+                "title": r"$T_{rad}\ [K]$",
+                "tickmode": "auto",
+                "nticks": 15,
+            },
+            yaxis2={
+                "tickformat": "g",
+                "title": r"$W$",
+                "tickmode": "auto",
+                "nticks": 15,
```

Comment:
Is it possible to add some of these things, like nticks, height, etc., as kwargs? Or some method of having this be user configurable

Reply:
The `height` and many other layout properties can be changed using the `plasma_plot_config` and the `luminosity_plot_config` options, but I'm afraid I didn't get the desired results using that method only for `nticks`. Apart from `nticks`, I believe most of the properties can be overridden. Setting `nticks` to say 10 will sometimes not create 10 `nticks`, but it does increase them, arbitrarily. I am working on it and will create an issue in plotly if I am able to recreate it.